mirai-core/src/commonMain/kotlin/contact/FriendImpl.kt

```diff
@@ -65,6 +66,13 @@ internal class FriendImpl(
     @Suppress("unused") // bug
     val lastMessageSequence: AtomicInt = atomic(-1)
 
+    override suspend fun delete(): Boolean {
+        return bot.network.run {
```

Comment:
应该在服务器给出错误之前先检查 this 仍然是 Bot 的好友:
```suggestion
        check(bot.friends[this.id] != null) {
            "Friend ${this.id} had already been deleted"
        }

        return bot.network.run {
```